modules/services/window-managers/labwc/labwc.nix

```diff
@@ -0,0 +1,272 @@
+      ];
+    };
+
+    environment = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = ''
+        Environment variable to add when labwc start.
+      '';
+      example = [
+        "XDG_CURRENT_DESKTOP=labwc:wlroots"
+        "XKB_DEFAULT_LAYOUT=us"
+      ];
+    };
+
+    systemd = {
+      enable = lib.mkEnableOption null // {
+        default = true;
+        description = ''
+          Whether to enable {file}`labwc-session.target` on
+          labwc startup. This links to {file}`graphical-session.target`.
+          Some important environment variables will be imported to systemd
+          and D-Bus user environment before reaching the target, including
+          - `DISPLAY`
+          - `WAYLAND_DISPLAY`
+          - `XDG_CURRENT_DESKTOP`
+        '';
+      };
+
+      variables = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [
+          "DISPLAY"
+          "WAYLAND_DISPLAY"
+          "XDG_CURRENT_DESKTOP"
+        ];
+        example = [ "-all" ];
+        description = ''
+          Environment variables to be imported in the systemd & D-Bus user
+          environment.
+        '';
+      };
+
+      extraCommands = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [
+          "systemctl --user stop labwc-session.target"
+          "systemctl --user start labwc-session.target"
+        ];
+        description = "Extra commands to be run after D-Bus activation.";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "wayland.windowManager.labwc" pkgs lib.platforms.linux)
+    ];
+
+    home.packages = lib.mkIf (cfg.package != null) (
+      [ cfg.package ] ++ lib.optional cfg.xwayland.enable pkgs.xwayland
+    );
+
+    xdg.configFile."labwc/rc.xml".text = function.generateXML "labwc_config" cfg.rc cfg.extraConfig;
+
+    xdg.configFile."labwc/menu.xml".text = function.generateXML "openbox_menu" cfg.menu "";
+
+    xdg.configFile."labwc/autostart".source = pkgs.writeShellScript "autostart" (
+      ''
+        ### This file was generated with Nix. Don't modify this file directly.
+
+        ### AUTOSTART SERVICE ###
+        ${lib.concatStringsSep "\n" cfg.autostart}
+
+      ''
+      + (lib.optionalString cfg.systemd.enable ''
+        ### SYSTEMD INTEGRATION ###
+        ${systemdActivation}
+      '')
+    );
+
+    xdg.configFile."labwc/environment".text = lib.concatStringsSep "\n" (
+      cfg.environment ++ (lib.optionals (!cfg.xwayland.enable) [ "WLR_XWAYLAND=" ])
+    );
+
+    systemd.user.targets.labwc-session = lib.mkIf cfg.systemd.enable {
+      Unit = {
+        Description = "labwc compositor session";
+        Documentation = [ "man:systemd.special(7)" ];
+        BindsTo = [ "graphical-session.target" ];
+        Wants = [ "graphical-session-pre.target" ];
+        After = [ "graphical-session-pre.target" ];
```

Comment:
Can you change these to follow the other wayland services that use `config.wayland.systemd.target`